cpp/src/Slice/Parser.cpp

```diff
@@ -4465,6 +4450,17 @@ Slice::Parameter::isOutParam() const
     return _isOutParam;
 }
 
+void
+Slice::Parameter::setIsOutParam()
+{
+    // Emit an error if this parameter was already marked as 'out' (meaning 'out' was written multiple times in a row).
+    if (_isOutParam)
+    {
+        unit()->error("parameter '" + _name + "' was marked 'out' multiple times");
```

Comment:
Does this emit a line error? Maybe we should add this to the error detection tests.